src/Testcontainers.WebDriver/WebDriverBuilder.cs

```diff
@@ -0,0 +1,228 @@
+    /// </summary>
+    /// <param name="timeZone">The desirable time zone.</param>
+    /// <returns>A configured instance of <see cref="WebDriverBuilder" />.</returns>
+    public WebDriverBuilder SetTimeZone(string timeZone)
+    {
+        return WithEnvironment("TZ", timeZone);
+    }
+
+    /// <summary>
+    /// Sets grid configuration by toml file.
+    /// </summary>
+    /// <remarks>
+    /// https://www.selenium.dev/documentation/grid/configuration/toml_options/
+    /// </remarks>
+    /// <param name="configTomlFilePath">The config toml file path.</param>
+    /// <returns>A configured instance of <see cref="WebDriverBuilder" />.</returns>
+    public WebDriverBuilder SetConfigurationFromTomlFile(string configTomlFilePath)
+    {
+        return WithResourceMapping(configTomlFilePath, "/opt/bin/config.toml");
+    }
+
+    /// <summary>
+    /// Sets ffmpeg video recording container.
+    /// </summary>
+    /// <remarks>
+    /// https://github.com/SeleniumHQ/docker-selenium#video-recording
+    /// </remarks>
+    /// <param name="fileName">video file name - default video name</param>
+    /// <param name="fileType">video file type - default mp4 type</param>
+    /// <param name="videosFolder">source of video folder on your host - default /tmp/videos path</param>
+    /// <returns>A configured instance of <see cref="WebDriverBuilder" />.</returns>
+    public WebDriverBuilder WithRecording(string fileName = "video", string fileType = "mp4", string videosFolder = "/tmp/videos")
+    {
+        var webDriverContainerNetworkName = Guid.NewGuid().ToString("D");
+
+        var network = new NetworkBuilder()
+            .WithDriver(NetworkDriver.Bridge)
+            .Build();
+
+        var recordingContainer = new ContainerBuilder()
+            .WithImage(WebDriverType.Video.Image)
+            .WithNetwork(network)
+            .WithEnvironment("FILE_NAME", $"{fileName}.{fileType}")
+            .WithEnvironment("DISPLAY_CONTAINER_NAME", webDriverContainerNetworkName)
+            .WithBindMount(videosFolder, "/videos")
+            .Build();
+
+        return Merge(DockerResourceConfiguration, new WebDriverConfiguration(network: network, recordingContainer: recordingContainer))
+            .WithNetwork(network)
+            .WithNetworkAliases(webDriverContainerNetworkName);
+    }
+
+    /// <inheritdoc />
+    public override WebDriverContainer Build()
+    {
+        Validate();
+        return new WebDriverContainer(DockerResourceConfiguration, TestcontainersSettings.Logger);
+    }
+
+    /// <inheritdoc />
+    protected override WebDriverBuilder Init()
+    {
+        return base.Init()
+            .WithBrowser(WebDriverType.Chrome)
+            .WithPortBinding(WebDriverPort, true)
+            .WithPortBinding(VncServerPort, true)
+            .WithWaitStrategy(Wait.ForUnixContainer().UntilPortIsAvailable(VncServerPort))
+            .WithWaitStrategy(Wait.ForUnixContainer().UntilHttpRequestIsSucceeded(request
```

Comment:
Your 2nd wait strategy overrides the 1st. I guess we can remove `.WithWaitStrategy(Wait.ForUnixContainer().UntilPortIsAvailable(VncServerPort))`.

Reply:
I think also
I will remove it